fix: Reflect updated Sequelize.AbstractQueryInterface (#369)
sequelize/website: docs/other-topics/upgrade.md

- A few bugs in how indexes were named have been fixed. This means your index names could change.

## Minor Breaking changes

> ### Renamed APIs

Reviewer: To keep things breathing :)

```suggestion

### Renamed APIs

```

> - `Sequelize.QueryInterface` has been renamed to `Sequelize.AbstractQueryInterface`. Previously, you could do: 

Reviewer: I wouldn't add the `Sequelize.` prefix. We don't want people to access these values on the constructor, that's just there for legacy reasons

Since another PR has been merged, let's add `ModelColumnAttributeOptions` to the list

```suggestion
The following exports have been renamed:

- `QueryInterface` has been renamed to `AbstractQueryInterface`.
- `ModelColumnAttributeOptions` has been renamed to `AttributeOptions`.
```

```javascript

> const { QueryInterface } = require('@sequelize/core');

Reviewer 2: ```suggestion
const { QueryInterface } = require('sequelize');
```

> ```javascript
> const { QueryInterface } = require('@sequelize/core');
> ```
> The new import would be:
> ```javscript
> const { AbstractQueryInterface } = require('@sequelize/core');
> ```

Reviewer: I don't know if this example is necessary, considering there is a good number of APIs that have been renamed (many of which are yet to be documented), but if you do add it:

- prefer `import` to `require`
- added a bit of spacing
- I used `sequelize` for the old package, since that's what v6 uses

````suggestion

```javascript
import { QueryInterface } from 'sequelize';
```

The new import would be:

```javascript
import { AbstractQueryInterface } from '@sequelize/core';
```
````

Author: Sure - I'll remove the example
